### DISCUSSION TITLE
How to merge into existing ApolloServer?

### DISCUSSION THREAD

**Discussion by theogravity** [hayes/pothos#167] [Q&A] [answered]
I currently have a project that does the following:

```
  return new ApolloServerExpress({
    typeDefs,
    schemaDirectives,
    resolvers,
    ...
})
```

I originally had the idea of using `mergeSchemas` to merge the schemas together, but found out that we do the above instead of using the `schema` property.

Is there a way to merge in my `giraphql` schema with that kind of config?

**Comment by theogravity** [comment, marked as answer]
Figured it out, import `makeExecutableSchema` and `mergeSchemas` from `apollo-server-express` / graphql-tools (if not using express)

```
  const schema = makeExecutableSchema({
    typeDefs,
    resolvers,
    schemaDirectives,
  });

  return new ApolloServerExpress({
    schema: mergeSchemas({
      schemas: [schema, giraphBuilder.toSchema({})],
    }),
    ```

**Comment by tslater** [comment]
I'm using the Relay plugin and it is totally colliding with the relay definitions in the other schema and causing weird issues. Anyone else have any experience working through that?

**Reply by theogravity** [reply]
One tip would be if you have a type defined in your original GQL server, and you use that in a merged with Pothos scenario, you also need to define the same type in Pothos as well or the merge won't work.

Ex you have a 'User' type defined in your GQL and you use it as a return type in Pothos, you need to create a Pothos version of User copying the same fields / requirements as the one in your GQL server.

I don't know why it works like that but we've had to do this duplication several times for types and enums until we can get rid of the usage of that type in the original GQL impls.
